de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/LemmaTechnologySubGenerator.xtend

```diff
@@ -0,0 +1,181 @@
+package de.fhdo.lemma.service.open_api
+
+import io.swagger.v3.oas.models.OpenAPI
+import de.fhdo.lemma.technology.TechnologyFactory
+import de.fhdo.lemma.technology.Technology
+import org.slf4j.LoggerFactory
+import de.fhdo.lemma.data.DataFactory
+import de.fhdo.lemma.technology.JoinPointType
+import de.fhdo.lemma.technology.AspectFeature
+import de.fhdo.lemma.data.PrimitiveType
+import de.fhdo.lemma.technology.TechnologySpecificPrimitiveType
+import io.swagger.v3.oas.models.PathItem
+import io.swagger.v3.oas.models.Operation
+import java.util.Set
+import de.fhdo.lemma.technology.ServiceAspect
+
+
+/**
+ * This class is responsible for handling the generation of a LEMMA technology model for the OpenAPI
+ * technology from an OpenAPI file in the JSON or YAML format.
+ *
+ * @author <a href="mailto:jonas.sorgalla@fh-dortmund.de">Jonas Sorgalla</a>
+ */
+class LemmaTechnologySubGenerator {
+    /** Factory to actually create and manipulate a LEMMA Technology */
+    val TECHNOLOGY_FACTORY = TechnologyFactory.eINSTANCE
+    /** Factory to actually create and manipulate a LEMMA DataModel */
+    val DATA_FACTORY = DataFactory.eINSTANCE
+    /** Predefined instance of the <strong>TechnologyModel</strong>. This instance is populated
+     * with the various technologies from the OpenAPI model, e.g., media types.
+     */
+    val myTechModel = TECHNOLOGY_FACTORY.createTechnology
+    /** OpenAPI schema which will be used as source for generation. */
+    OpenAPI openApi
+    /** SLF4j Logger */
+    val static logger = LoggerFactory.getLogger(LemmaServiceSubGenerator);
+    /** Location where the generated file is written */
+    String targetFolder
+    /** List of all unique media types. */
+    val mediaTypes = <String>newHashSet
+
+    new(OpenAPI api, String genPath, String techFilename) {
+        super()
+        logger.debug("Creating new Technology Sub Generator...")
+        this.openApi = api
+        this.targetFolder = genPath + techFilename
+    }
+
+    def Technology generate() {
+        logger.debug("Starting generation...")
+        logger.debug("Initializing model instance...")
+        initilize()
+        logger.debug("...Technology Model initialized!")
+        logger.debug("Creating protocols...")
+        createProtocols()
+        logger.debug("...Data Structures created!")
+
+        if (OpenApiUtil.writeModel(myTechModel, targetFolder)) {
+            logger.info("Technology model generation successful!")
+            logger.info('''Model written to «targetFolder»''')
+        } else
+            throw new Exception("Technology model generation failed :(")
+        return myTechModel
+    }
+
+    def createProtocols() {
+        //creates "rest" as the default openapi protocol
+        val rest = TECHNOLOGY_FACTORY.createProtocol
+        rest.name = "rest"
+        openApi.paths.forEach[path, pathSchema|searchMediaTimes(pathSchema)]
```

Comment:
It would be way better to clearly communicate the manipulation of the `mediaTypes` attribute at this place instead of in the `searchMediaTimes" (typo btw) method. Something like the following would be more clear what is going on here:
```
mediaTypes = findAllMediaTypesFrom(openApiSpec)
```
and then have `findAllMediaTypesFrom` handle the iteration over the paths of the given `openApiSpec`. Even more, why can't you just initialize the attribute immediately upon its definition:
```
val mediaTypes = findAllMediaTypesFrom(openApiSpec)
```
?